python/generateTemplate.py

```diff
@@ -0,0 +1,1155 @@
+            If not set, uses the existing self.butler
+        band_name : str, optional
+            Common name of the filter used. For LSST, use u, g, r, i, z, or y
+        **kwargs :
+            Any additional keyword arguments to pass to ``load_bandpass``
+
+        Returns
+        -------
+        None, but loads self.model and sets up all the needed quantities such as the psf and bandpass objects.
+        """
+        self.instrument = instrument
+        self.filter_name = band_name
+        model_arr = []
+        dcrModel_gen = self.read_exposures(data_type="dcrModel", input_repository=model_repository)
+        for dcrModel in dcrModel_gen:
+            model_arr.append(dcrModel.getMaskedImage().getImage().getArray())
+
+        self.model = model_arr
+        # The weights should be identical for all subfilters.
+        self.weights = dcrModel.getMaskedImage().getVariance().getArray()
+        # The masks should be identical for all subfilters
+        self.mask = dcrModel.getMaskedImage().getMask().getArray()
+
+        self.wcs = dcrModel.getWcs()
+        self.n_step = len(self.model)
+        y_size, x_size = dcrModel.getDimensions()
+        self.x_size = x_size
+        self.y_size = y_size
+        self.pixel_scale = self.wcs.pixelScale()
+        self.exposure_time = dcrModel.getInfo().getVisitInfo().getExposureTime()
+        self.observatory = dcrModel.getInfo().getVisitInfo().getObservatory()
+        self.bbox = dcrModel.getBBox()
+        bandpass_init = self.load_bandpass(band_name=band_name, **kwargs)
+        wavelength_step = (bandpass_init.wavelen_max - bandpass_init.wavelen_min) / self.n_step
+        self.bandpass = self.load_bandpass(band_name=band_name, wavelength_step=wavelength_step, **kwargs)
+
+        self.psf = dcrModel.getPsf()
+        psf_avg = self.psf.computeKernelImage().getArray()
+        self.psf_size = psf_avg.shape[0]
+        self.debug = False
+
+    @staticmethod
+    def _calc_offset_phase(dcr_gen, exposure=None, size=None, size_out=None, center_only=False):
+        """Calculate the covariance matrix for a simple shift with no psf.
+
+        Parameters
+        ----------
+        dcr_gen : generator
+             A dcr generator of offsets, returned by ``_dcr_generator``.
+        exposure : lsst.afw.image.ExposureD object, optional
+            An LSST exposure object. Only needed if size is not specified.
+        size : int, optional
+            Width in pixels of the region used in the origin image. Default is entire image
+        size_out : int, optional
+            Width in pixels of the region used in the destination image. Default is same as size
+        center_only : bool, optional
+            Set to True to calculate the covariance for only the center pixel.
+
+        Returns
+        -------
+        np.ndarray
+            Returns the covariance matrix of an offset generated by _dcr_generator in the form (dx, dy)
```

Comment:
I don't know what "in the form (dx, dy)" means. Would it be clear to somebody more familiar with the concepts?